Controllers/user.js

```diff
@@ -0,0 +1,76 @@
+const Address = require('../Database/address');
+const Cart = require('../Database/cart');
+const User = require('../Database/user');
+const Wishlist = require('../Database/wishlist');
+var jwt = require('jsonwebtoken');
+
+exports.signupHandler = async (req, res) => {
+	const user = new User(req.body);
+	try {
+		const token = jwt.sign({ userId: user._id }, 'secret');
+		const savedUser = await user.save();
+		const userCart = new Cart({
+			_id: savedUser._id,
+		});
+		const savedCart = await userCart.save();
+		const userWishlist = new Wishlist({
+			_id: savedUser._id,
+		});
+		const savedWishlist = await userWishlist.save();
+		const userAddress = new Address({
+			_id: savedUser._id,
+		});
+		const savedUserAddress = await userAddress.save();
+		await Cart.findOne({ _id: savedCart._id })
+			.populate('_id')
+			.exec(function (err, usercart) {
+				console.log('The author is %s', usercart);
+			});
+
+		await Wishlist.findOne({ _id: savedWishlist._id })
+			.populate('_id')
+			.exec(function (err, userWishlist) {
+				console.log('The author is %s', userWishlist);
+			});
+
+		await Address.findOne({ _id: savedUserAddress._id })
+			.populate('_id')
+			.exec(function (err, userAddress) {
+				console.log('The author is %s', userAddress);
+			});
+
+		res.json({
+			token,
+			user: savedUser,
```

Comment:
If I am not missing anything, savedUser will have _id and password as well, please remove it before sending savedUser in response
```
savedUser._id = undefined
savedUsed.password = undefined
```
Also if possible remove user id's from other objects as well before sending in response for data security.

Please use bcrypt+salt for encrypting password, before savedUser.save()

Reply:
Done, changed the entire code for userlogin and signup